controllers/articles-controllers.js

```diff
@@ -17,8 +18,17 @@ function getArticleById(request, response, next) {
 }
 
 function getArticles(request, response, next) {
-  const { sort_by, order } = request.query;
-  selectArticles(sort_by, order)
+  const { sort_by, order, topic } = request.query;
+  let articlePromise;
+  if (topic) {
+    articlePromise = selectArticleByTopic(topic).then(() => {
+      return selectArticles(sort_by, order, topic);
+    });
+  } else {
+    articlePromise = selectArticles(sort_by, order, topic);
```

Comment:
This works, but could be DRY-er, think about how you could amend your selectArticles model so it can respond whether it has a topic or not.